src/inmanta/server/services/orchestrationservice.py

```diff
@@ -943,7 +953,17 @@ async def _put_version(
                         raise BadRequest(msg)
                     all_ids |= {Id.parse_id(rid, version) for rid in rids_unchanged_resource_sets.keys()}
 
-                await data.Resource.insert_many(list(rid_to_resource.values()), connection=connection)
+                updated_resources = list(rid_to_resource.values())
+                await data.Resource.insert_many(updated_resources, connection=connection)
+                # bump all resource sets if we are doing a full compile otherwise, bump only the updated resource sets
+                if not is_partial_update:
+                    updated_resource_sets = {r.resource_set if r.resource_set is not None else "" for r in updated_resources}
```

Comment:
Up to now, this method (and the rest of the code iirc) used `None` to indicate the default resource set. With this new implementation this method uses a mix of two styles: 1) None for the default resource set or 2) the empty string for the default resource set. I think we have to keep it consistent, otherwise we will easily get bugs. If we use a different style in the db compared to the code, that's fine for me. But in the code we should try to keep it consistent.

In fact a bug like this is already introduced on the `resource_sets` variable. This dict uses None to refer to the default resource set. The new code you added at the beginning of this method uses the empty string.

Reply:
I agree that it is confusing and I would prefer not to have this None/empty string issue. I think I do not have a bug here.
In both cases I iterate over a list of resources so I have to make the conversion from None to ""